### PR TITLE
[Enhancement] Throw compile error when _GLIBCXX_USE_CXX11_ABI=0

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

 Modifying `SinkBuffer::buffer `is protected by mutex, but getting size of buffer isn't protected to gain better performance. `std::list::size` before gcc5 traverses the whole list, which is not thread-safe.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

File: be/src/exec/pipeline/exchange/sink_buffer.cpp [hunk 1]
```diff
@@ -86,7 +76,13 @@ Status SinkBuffer::add_request(TransmitChunkInfo& request) {
     }
     {
         auto& instance_id = request.fragment_instance_id;
-        RETURN_IF_ERROR(_try_to_send_rpc(instance_id, [&]() { _buffers[instance_id.lo].push(request); }));
+        RETURN_IF_ERROR(_try_to_send_rpc(instance_id, [&]() {
+            auto& item = _instance_id_lo2item[instance_id.lo];
+#if !_GLIBCXX_USE_CXX11_ABI
```
Comment (satanson) [inline be/src/exec/pipeline/exchange/sink_buffer.cpp:81]: suggest to wrap a pair of std::size_t, std::list into a new type list, for ex example:

```cpp
template<typename T>
class SizedList {
public:
   push(...){
     _list.push(...);
#if !_GLIBCXX_USE_CXX11_ABI
     _size.fetch_add(1);
#endif
   }
....
private:
std::list<T> _list;
#if !_GLIBCXX_USE_CXX11_ABI
std::atomic<size_t> _size
#endif
}
```

**Review by satanson — DISMISSED** [review]

**Review by murphyatwork — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by liuyehcf — APPROVED** [review]

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
@Mergify backport branch-2.2 branch-2.3 branch-2.4 branch-2.5

**Pull request merged**